### PR TITLE
Remove InstrumentRecorder class

### PR DESCRIPTION
**Pull request by tarekgh**
Removing InstrumentRecorder from the runtime. [MetricCollector](https://github.com/dotnet/extensions/blob/main/src/Libraries/Microsoft.Extensions.Telemetry.Testing/Metering/MetricCollector.cs) can be used instead.

### CODE REVIEW

**Comment by tarekgh** [comment]
CC @geeknoid @stephentoub @noahfalk @JamesNK @joperezr @ericstj

**Review by joperezr — APPROVED** [review]
LGTM. Thanks Tarek.

**Review by geeknoid — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Diagnostics/tests/MetricsTests.cs [hunk 1]
```diff
@@ -116,69 +116,35 @@ public void MeterDisposeTest()
             IMeterFactory meterFactory = sp.GetRequiredService<IMeterFactory>();
 
             Meter meter = meterFactory.Create("DisposableMeter");
-
             Counter<int> counter = meter.CreateCounter<int>("MyCounter");
-            InstrumentRecorder<int> recorder = new InstrumentRecorder<int>(counter);
+
+            using MeterListener listener = new MeterListener();
+            listener.InstrumentPublished = (instrument, theListener) =>
```
Comment (noahfalk) [inline src/libraries/Microsoft.Extensions.Diagnostics/tests/MetricsTests.cs:122]: This works and I'm fine if you merge it this way, but a simpler option is:

```C#
using MeterListener listener = new MeterListener();
int lastMeasurement = 0;
listener.SetMeasurementEventCallback<int>((inst, measurement, tags, state) => lastMeasurement = measurement);
listener.EnableMeasurementEvents(counter, null);
```
There is no requirement to use publishing events if you already know the counter you want to listen to.

**Review by noahfalk — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.Diagnostics/tests/MetricsTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Diagnostics/tests/MetricsTests.cs:122]: I'll fix this in another PR. Thanks!

**Review by ericstj — APPROVED** [review]
LGTM since `InstrumentRecorder` was added in P5 it's OK to remove now.  You may want to issue a breaking change notice, just in case any customer's started using it in the preview.

**Comment by tarekgh** [comment]
> You may want to issue a breaking change notice, just in case any customer's started using it in the preview.

I can add something to https://github.com/dotnet/core/issues/8437

File: src/libraries/Microsoft.Extensions.Diagnostics/tests/MetricsTests.cs (hunk 1 above)
Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.Diagnostics/tests/MetricsTests.cs:122]: opened the PR https://github.com/dotnet/runtime/pull/87898

**Pull request merged**